src/ring_theory/dedekind_domain/ideal.lean

```diff
@@ -847,35 +845,64 @@ variables [is_domain R] [is_dedekind_domain R]
 
 /-- The height one prime spectrum of a Dedekind domain `R` is the type of nonzero prime ideals of
 `R`. Note that this equals the maximal spectrum if `R` has Krull dimension 1. -/
-@[ext, nolint has_nonempty_instance unused_arguments]
-structure height_one_spectrum :=
-(as_ideal : ideal R)
-(is_prime : as_ideal.is_prime)
-(ne_bot   : as_ideal ≠ ⊥)
+@[nolint has_nonempty_instance unused_arguments]
+def height_one_spectrum := {I : ideal R // I.is_prime ∧ I ≠ ⊥}
 
 variables (v : height_one_spectrum R) {R}
 
-lemma height_one_spectrum.prime (v : height_one_spectrum R) : prime v.as_ideal :=
-ideal.prime_of_is_prime v.ne_bot v.is_prime
+namespace height_one_spectrum
 
-lemma height_one_spectrum.irreducible (v : height_one_spectrum R) :
-  irreducible v.as_ideal :=
-begin
-  rw [unique_factorization_monoid.irreducible_iff_prime],
-  apply v.prime,
-end
+/-- View a point in the height one spectrum of a commutative ring as an ideal of that ring. -/
+abbreviation as_ideal (x : height_one_spectrum R) : ideal R := x.val
+
+@[ext] lemma ext {x y : height_one_spectrum R} : x = y ↔ x.as_ideal = y.as_ideal :=
+subtype.ext_iff_val
+
+instance is_prime : v.as_ideal.is_prime := v.property.left
+
+lemma ne_bot : v.as_ideal ≠ ⊥ := v.property.right
+
+instance is_maximal : v.as_ideal.is_maximal := dimension_le_one v.as_ideal v.ne_bot v.is_prime
+
+lemma prime : prime v.as_ideal := ideal.prime_of_is_prime v.ne_bot v.is_prime
 
-lemma height_one_spectrum.associates_irreducible (v : height_one_spectrum R) :
-  irreducible (associates.mk v.as_ideal) :=
+lemma irreducible : irreducible v.as_ideal :=
+unique_factorization_monoid.irreducible_iff_prime.mpr v.prime
+
+lemma associates_irreducible : _root_.irreducible $ associates.mk v.as_ideal :=
+(associates.irreducible_mk _).mpr v.irreducible
+
+/-- An equivalence between the height one and maximal spectra for rings of Krull dimension 1. -/
+def equiv_maximal_spectrum (hR : ¬is_field R) : height_one_spectrum R ≃ maximal_spectrum R :=
+{ to_fun    := λ ⟨v, hv, hbot⟩, ⟨v, dimension_le_one v hbot hv⟩,
+  inv_fun   := λ ⟨v, hv⟩, ⟨v, hv.is_prime, ring.ne_bot_of_is_maximal_of_not_is_field hv hR⟩,
```

Comment:
```suggestion
{ to_fun    := λ v, ⟨v, dimension_le_one v v.2.2 v.2.1⟩,
  inv_fun   := λ v, ⟨v, v.2.is_prime, ring.ne_bot_of_is_maximal_of_not_is_field v.2 hR⟩,
```
It is preferred not to destruct the argument when constructing data.

Note: suggestion not tested locally.